crates/lib3h/src/engine/network_layer.rs

```diff
@@ -188,7 +190,11 @@ impl<T: Transport, D: Dht> RealEngine<T, D> {
                 let space_gateway = self
                     .space_gateway_map
                     .get_mut(&(msg.space_address.to_owned(), msg.to_peer_address.to_owned()))
-                    .ok_or_else(|| format_err!("space_gateway not found"))?;
+                    .ok_or_else(|| {
+                        Lib3hError::new(ErrorKind::KeyNotFound(String::from(
+                            "space_gateway not found",
+                        )))
+                    })?;
```

Comment:
This seems way to verbose for me.
Can we have a follow-up PR with converters or a macro that would allow us to write something link:
`lib3h_err!(KeyNotFound, ""space_gateway not found")` ?

Reply:
Yeah I agree, this is kinda too verbose and that's on my list of improvement :)